src/util/misc.rs

```diff
@@ -144,8 +147,9 @@ mod message_signing {
             &self,
             secp_ctx: &secp256k1::Secp256k1<C>,
             msg_hash: sha256d::Hash
-        ) -> Result<PublicKey, secp256k1::Error> {
-            let msg = secp256k1::Message::from_slice(&msg_hash[..])?;
+        ) -> Result<PublicKey, MessageSignatureError> {
+            let msg = secp256k1::Message::from_slice(&msg_hash[..])
+                .expect("cannot fail");
```

Comment:
Actually just  `let msg = secp256k1::Message::from(msg_hash);` [should be possible](https://docs.rs/secp256k1/latest/secp256k1/struct.Message.html#impl-From%3CT%3E) if we [activate](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/Cargo.toml#L39) `bitcoin_hashes` feature of secp256k1 .

Reply:
Oh nice, I forgot that we finally added that feature. Yes, let's do this.

Reply:
Opened #824 for that.

Reply:
Nice catch. I was looking for the fewest changes possible but this is definitely an improvement.

@apoelstra @Kixunil I pushed the changes in 8a05ca042d4e8